test/distributed/org/apache/cassandra/distributed/test/ReadCoordinationMetricsTest.java

```diff
@@ -0,0 +1,124 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.distributed.test;
+
+import java.net.InetAddress;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import org.apache.cassandra.distributed.Cluster;
+import org.apache.cassandra.distributed.api.IInvokableInstance;
+import org.apache.cassandra.distributed.api.IIsolatedExecutor;
+import org.apache.cassandra.distributed.impl.DistributedTestSnitch;
+import org.apache.cassandra.locator.InetAddressAndPort;
+import org.apache.cassandra.metrics.ReadCoordinationMetrics;
+import org.apache.cassandra.service.reads.AbstractReadExecutor;
+
+import static org.apache.cassandra.distributed.api.ConsistencyLevel.ALL;
+import static org.apache.cassandra.distributed.api.ConsistencyLevel.ONE;
+
+public class ReadCoordinationMetricsTest extends TestBaseImpl
+{
+    private static long countNonreplicaRequests(IInvokableInstance node)
+    {
+        return node.callOnInstance(() -> ReadCoordinationMetrics.nonreplicaRequests.getCount());
+    }
+
+    private static long countPreferredOtherReplicas(IInvokableInstance node)
+    {
+        return node.callOnInstance(() -> ReadCoordinationMetrics.preferredOtherReplicas.getCount());
+    }
+
+    /**
+     * Two nodes with RF=1 so half the data will be owned by each node and the coordinator for queries is not
+     * always a replica in the list of candidates in {@link AbstractReadExecutor#getReadExecutor} where
+     * {@link ReadCoordinationMetrics#nonreplicaRequests} will be incremented.
+     *
+     * @throws Throwable
+     */
+    @Test
+    public void testNonReplicaRequests() throws Throwable
+    {
+        try (Cluster cluster = init(Cluster.create(2), 1))
+        {
+            cluster.schemaChange(withKeyspace("CREATE TABLE %s.tbl (pk int, ck int, v int, PRIMARY KEY (pk, ck))"));
+            for (int i = 0; i < 100; i++)
```

Comment:
Since the magic number `100` is number of rows inserted, queried and expected, will it be good to put it into a constant?